.dotnet/tests/Utility/MockPipelineResponse.cs

```diff
@@ -160,6 +160,7 @@ public override async ValueTask<BinaryData> BufferContentAsync(CancellationToken
 
         // Less efficient FromStream method called here because it is a mock.
         // For intended production implementation, see HttpClientTransportResponse.
+        _contentStream.Position = 0;
```

Comment:
You have no idea how much time it took me to figure out I had to add this line to make async tests work.

Reply:
I wouldn't have known either :/